packages/infra/lib/ihe-gateway-v2-stack.ts

```diff
@@ -38,6 +44,54 @@ export class IHEGatewayV2LambdasNestedStack extends NestedStack {
     patientDiscoveryLambda.grantInvoke(props.apiService.taskDefinition.taskRole);
     documentQueryLambda.grantInvoke(props.apiService.taskDefinition.taskRole);
     documentRetrievalLambda.grantInvoke(props.apiService.taskDefinition.taskRole);
+
+    this.addMetricFiltersAndAlarms(patientDiscoveryLambda, "PatientDiscoveryLambda", props, [
+      { filterPattern: "Aborted Error In Soap Response", threshold: 100 },
+      { filterPattern: "Failure Sending SAML Request", threshold: 100 },
+    ]);
+
+    this.addMetricFiltersAndAlarms(documentQueryLambda, "DocumentQueryLambda", props, [
+      { filterPattern: "RegistryErrorList In Soap Response", threshold: 5 },
+      { filterPattern: "Failure Sending SAML Request", threshold: 5 },
+    ]);
+
+    this.addMetricFiltersAndAlarms(documentRetrievalLambda, "DocumentRetrievalLambda", props, [
+      { filterPattern: "RegistryErrorList In Soap Response", threshold: 5 },
+      { filterPattern: "Failure Sending SAML Request", threshold: 5 },
+    ]);
+  }
+
+  private addMetricFiltersAndAlarms(
+    lambdaFunction: Lambda,
+    functionName: string,
+    props: IHEGatewayV2LambdasNestedStackProps,
+    filters: { filterPattern: string; threshold: number }[]
+  ) {
+    const logGroup = lambdaFunction.logGroup;
+
+    filters.forEach(({ filterPattern, threshold }) => {
+      const sanitizedFilterPattern = filterPattern.replace(whitespaceRegex, "");
+      const metricFilter = logGroup.addMetricFilter(
+        `${functionName}-${sanitizedFilterPattern}-MetricFilter`,
+        {
+          metricNamespace: "IHEGatewayV2",
+          metricName: `${functionName}-${sanitizedFilterPattern}`,
+          filterPattern: FilterPattern.anyTerm(filterPattern),
```

Comment:
Won't this result in us getting hits for log entries that contain _any_ of the components of the filter?